distribution/lib/Standard/Test/0.0.0-dev/src/Suite.enso

```diff
@@ -41,59 +41,65 @@ type Suite_Builder
                 self.builder.append group
 
 
-
 ## A testing suite that holds all the groups with test specs.
 
    Users are expected to build the suite via `Suite_Builder`. That is, via `Test.build` method.
 type Suite
     ## PRIVATE
     Impl (groups : Vector Group)
 
-    ## Run only tests that match the provided filters.
-
-       A filter can be either Regex, Text, or Nothing. If Nothing, it matches all the names.
-       If Text, it matches only names that contains that text as a substring.
-       If a Regex, the `regex.match` method is invoked for every name.
-
-       Note that you can specify the regex such that it matches, e.g., exactly two test spec names
-       via something like `Regex.compile "(name1|name2)"`.
+    ## Run only tests that match the provided filter.
 
        Arguments:
-       - group_filter: Filter for group names.
-       - spec_filter: Filter for spec names.
-       - should_exit: If true, executes `System.exit` at the end.
-    run_with_filter : (Regex | Text | Nothing) -> (Regex | Text | Nothing) -> Boolean -> Nothing
-    run_with_filter self group_filter=Nothing spec_filter=Nothing should_exit=True =
-        config = Suite_Config.from_environment
+       - filter: Either text or Nothing. If Nothing, matches all the specs and all the groups.
+         If text, only groups or specs that exactly match the text are run.
```

Comment:
Could this be a regex? In the previous test framework I used to add a unique string to a test or group, and then use just the unique string as the filter. I know it's not much more work to copy an entire test name, but this is a chore that I do many times a day.

Reply:
+1

In the past we had the `TEST_ONLY_GROUP` env var and it accepting regex was quite a big help as it allowed more customization in which tests I want to run.

Reply:
> Could this be a regex?

Sure, you are the most heavy users. So shall the passed `filter` text parameter simply be interpreted as Regex? I wanted to avoid the previous complexities where I defined the parameter as `filter : (Regex | Text | Nothing)`.

Reply:
I think always treating it as regex is OK. After all `foo` is also a valid regex :)

So it would just be good to allow for not-full matches in the logic - i.e. regex `foo` should still match a group name `foobar`.

We could have some way to toggle whether it's regex or simple contains but that feels overcomplicated.

Reply:
Implemented in 783752b78a3732bcc2b3e186d9a48fdbf5e626b2. The `filter` argument is now treated as regex.

@GregoryTravis Your workflow should now work: Applying:
```diff
diff --git a/test/Base_Tests/src/Data/Vector_Spec.enso b/test/Base_Tests/src/Data/Vector_Spec.enso
index fba189b9d..d77af79e3 100644
--- a/test/Base_Tests/src/Data/Vector_Spec.enso
+++ b/test/Base_Tests/src/Data/Vector_Spec.enso
@@ -90,7 +90,7 @@ type_spec suite_builder name alter = suite_builder.group name group_builder->
         const = Vector.new 4 _->next
         const.should_equal [0, 1, 2, 3]
 
-    group_builder.specify "should allow vector creation with a constant constructor" <|
+    group_builder.specify "XXshould allow vector creation with a constant constructor" <|
         Vector.fill 100 1 . fold (0) (+) . should_equal 100
 
         r = Ref.new 0
```
And then running `enso --run test/Base_Tests/src/Data/Vector_Spec.enso XX` does:
```
Use a slice of an array as vectors: [1/1, 56ms]
    - XXshould allow vector creation with a constant constructor [56ms]
Use Array_Proxy as vectors: [1/1, 4ms]
    - XXshould allow vector creation with a constant constructor [4ms]
Use Vector as vectors: [1/1, 2ms]
    - XXshould allow vector creation with a constant constructor [2ms]
Use Array as vectors: [1/1, 2ms]
    - XXshould allow vector creation with a constant constructor [2ms]
Use Java ArrayList as vectors: [1/1, 2ms]
    - XXshould allow vector creation with a constant constructor [2ms]
5 tests succeeded.
0 tests failed.
0 tests skipped.
0 groups skipped.
```

That is, it matches all the specs that contain XX in all the groups in that file.

Reply:
Moreover, if the regex cannot be compiled:
```
$ enso --run test/Base_Tests/src/Data/Vector_Spec.enso '*********'
Execution finished with an error: Illegal Argument: Filter '*********' cannot be converted to regex
        at <enso> Panic.throw(Internal)
        at <enso> case_branch<arg-2>(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Test/0.0.0-dev/src/Suite.enso:128:17-107)
        at <enso> case_branch(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Test/0.0.0-dev/src/Suite.enso:127-128)
        at <enso> Suite.name_matches(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Test/0.0.0-dev/src/Suite.enso:124-129)
        at <enso> <anonymous>(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Test/0.0.0-dev/src/Suite.enso:66:29-58)
        at <enso> f(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Internal/Array_Like_Helpers.enso:243:22-48)
        at <enso> go<arg-2>(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso:312:31-50)
        at <enso> go(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso:311-313)
        at <enso> Range.fold<arg-2>(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso:314:13-30)
        at <enso> Range.fold(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso:308-314)
        at <enso> Array_Like_Helpers.fold(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Internal/Array_Like_Helpers.enso:244:5-39)
        at <enso> Vector.fold(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso:256:9-50)
        at <enso> Suite.run_with_filter(/home/pavel/dev/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Test/0.0.0-dev/src/Suite.enso:65-79)
        at <enso> Vector_Spec.main(Vector_Spec.enso:1088:5-32)
```

Reply:
This is great! ❤️ 🎉